### ISSUE TITLE
The source connector's Send Attempts and Response Date sometimes gets overwritten with incorrect values when source queue is enabled

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3194]
Haven't confirmed it for pre 3.0.3 versions but I'm pretty cure it will happen.

The response content of the source connector DOES get set, and they are all part of the same transaction. There is no exception. 

Originally thought it was related to MIRTH-3280 but seems like this also happens on Postgres even though Postgres does not deadlock. (WUT!?)

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3288
Reporter: wayneh
Created: 2014-05-19T17:21:50.000-0700

**Comment by rbeckman-nextgen** [comment]
LOL duh, it is being set, but there's a race condition going on because updateStatus also sets send_attempts and response_date. Basically the correct values are being overwritten. We'll probably have to make updateStatus NOT update these and create a different query for the destination connector to just set them apart from the status

Imported Comment. Original Details:
Author: wayneh
Created: 2014-05-19T17:50:08.000-0700

**Comment by rbeckman-nextgen** [comment]
If it helps, I can confirm it also happens in 3.0.2.7140 running with Postgres (this is a virtual appliance).

Imported Comment. Original Details:
Author: daigoba66
Created: 2014-05-21T08:13:57.000-0700

**Comment by rbeckman-nextgen** [comment]
Fixed the part where this was broken.

Imported Comment. Original Details:
Author: wayneh
Created: 2014-07-29T13:48:54.000-0700

**Comment by rbeckman-nextgen** [comment]
After many failed attempts at replicating, the below steps finally worked for me:

Created two channels, one with a tcp listener and the other with a jscript reader on the source and a tcp sender as the connector; the tcp listener had the source queue enabled.

The jscript reader was configured with a frequency of 1 millisecond so as to create a queue on the downstream source connector. After blasting the destination channel with messages, the originating channel was paused and the tcp listener channel refreshed a number of times. The sent attempts and response date were updated for the destination while nothing updated for the source. I exported and performed the same steps but was unable to replicate the issue. The channel correctly populated send attempts and response dates.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-09-02T14:28:09.000-0700
